modules/k8s/tunnel_test.go

```diff
@@ -0,0 +1,114 @@
+// +build kubeall kubernetes
+
+// NOTE: we have build tags to differentiate kubernetes tests from non-kubernetes tests. This is done because minikube
+// is heavy and can interfere with docker related tests in terratest. To avoid overloading the system, we run the
+// kubernetes tests separately from the others.
+
+package k8s
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+func TestTunnelOpensAPortForwardTunnelToPod(t *testing.T) {
+	t.Parallel()
+
+	uniqueID := strings.ToLower(random.UniqueId())
+	options := NewKubectlOptions("", "")
+	options.Namespace = uniqueID
+	configData := fmt.Sprintf(EXAMPLE_POD_YAML_TEMPLATE, uniqueID, uniqueID)
+	defer KubectlDeleteFromString(t, options, configData)
+	KubectlApplyFromString(t, options, configData)
+	WaitUntilPodAvailable(t, options, "nginx-pod", 60, 1*time.Second)
+
+	// Open a tunnel to pod from any available port locally
+	localPort := GetAvailablePort(t)
+	tunnel := NewTunnel(options, ResourceTypePod, "nginx-pod", localPort, 80)
```

Comment:
Nice, powerful API hiding a lot of complexity 👍

Reply:
Can't take credit for it since it is copied from `helm` 😄  But yes I like this API.